src/components/tx/modals/BatchExecuteModal/ReviewBatchExecute.tsx

```diff
@@ -24,12 +24,14 @@ import { useWeb3 } from '@/hooks/wallets/web3'
 const ReviewBatchExecute = ({ data, onSubmit }: { data: BatchExecuteData; onSubmit: (data: null) => void }) => {
   const [isSubmittable, setIsSubmittable] = useState<boolean>(true)
   const [submitError, setSubmitError] = useState<Error | undefined>()
+  const [executionMethod, setExecutionMethod] = useState(ExecutionMethod.RELAY)
   const chain = useCurrentChain()
   const { safe } = useSafeInfo()
   const [relays] = useRelaysBySafe()
 
   // Chain has relaying feature and available relays
-  const willRelay = relays && relays.remaining > 0
+  const canRelay = relays && relays.remaining > 0
+  const willRelay = canRelay && executionMethod === ExecutionMethod.RELAY
```

Comment:
Time to make an util for this logic? It will become less prone to inconsistencies